Moving Functions

cslotboom opened the pull request:
Here I move around some functions and imports. This will make it easier to track final changes to all the functions.

All of the functions  from "Get_Fiber Response" and "Get_Rendering_Animations" has been moved, and the files deleted.

Some functions will need to be imported now. I will fix it as I go through my final cleaning and making the test suite.

cslotboom committed:
Moving Functions

Here I move around some functions and inputs.

All of the functions  from "Get_Fiber Response" and "Get_Rendering_Animations" has been changed.

This breaks things! Some inpurts are now missing. I will fix it as I go through my final cleaning making the test suite the test functions.

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+    NodeText = [None]*Nnodes
+
+    # initialize figure
+    fig, ax = _initializeFig(nodes[:,1:], ndm)    
```
_initializeFig needs a link to the internal_plotting_functions, i.e. ipltf._initializeFig()

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+    fig, ax = _initializeFig(nodes[:,1:], ndm)    
+    
+	# Adjust plot area.   
+    _setStandardViewport(fig, ax, nodes[:,1:], ndm, Disp[0,:,:])
```
It should be ipltf._setStandardViewport()

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+        time_text = ax.text(0.95, 0.01, '', verticalalignment='bottom', 
+                            horizontalalignment='right', transform=ax.transAxes, color='grey')
+        
+        EQObjects = _plotEle_2D(nodes, elements, initialDisp, fig, ax, show_element_tags)
```
ipltf._plotEle_2D()

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+                
+    if ndm == 3:
+        
+        EQObjects = _plotEle_3D(nodes, elements, initialDisp, fig, ax, show_element_tags)
```
ipltf._plotEle3D()

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+    # Get displacements
+    if Plot_Displacements == 'yes':
+        # Get Node coordinants
+        OBD = readODB(ModelName, LoadCaseName)
```
Use idbf._readNodeDispData(Model,LoadCase) instead.

u-anurag replied:
You can check plot_deformedshape() function for reference.

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+    NodeText = [None]*Nnodes
+    
+    # Initialize figure
+    fig, ax = _initializeFig(DispNodeCoordArray, ndm)
```
it should be ipltf._initializeFig()

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+    if ndm == 2:
+        
+        # Plot elements
+        OutputObjects = _plotEle_2D(nodes, elements, DispNodeCoordArray, fig, ax, show_element_tags)
```
ipltf._plotEle_2D()

u-anurag commented on openseespy-pip/openseespy/postprocessing/Get_Rendering.py:
```diff
@@ -792,3 +798,618 @@ def nodecoordsFinal(nodetag):
+        print('3D model')
+		
+        # Plot Model and make Objects
+        OutputObjects = _plotEle_3D(nodes, elements, DispNodeCoordArray, fig, ax, show_element_tags)
```
ipltf._plotEle_3D()

u-anurag commented on openseespy-pip/openseespy/postprocessing/internal_database_functions.py:
```diff
@@ -293,3 +293,13 @@ def _readNodeDispData(ModelName,LoadCaseName):
+#### Read fibre data
+
+
+def recordFiberResponse(OutputName,eleNumber, sectionNumber):
```
Where does it save the file? If this function is available to users, shouldn't it be in Get_Rendering() ?

u-anurag reviewed (changes requested):
@cslotboom There are some comments. I assume you are running all the OpenSeesPy examples using the changed plotting files on your local machine to make sure they all work before you initiate a PR. If not, I would suggest to save all the plotting files in C-Moving_Functions branch to your local machine and run all the examples using them. That would be a better way to find all the missing links.

cslotboom committed:
Fixing Imports, updated fibre functions

Changed imports to work correctly
General bug fixing on plot functions.
Changed the fiber data to save to OBD
Made savefiberdata2D save to OBD by default, made function availible to users.

cslotboom committed:
Moved Save Fiber Data

u-anurag reviewed (approved):
Looks good. Please use idbf._readNodesandElements(ModelName) and idbf._readNodeDispData(ModelName,LoadCaseName) functions instead of readODB().
